StationeersLaunchPad/LaunchPadUpdater.cs

```diff
@@ -0,0 +1,138 @@
+﻿using Assets.Scripts;
+using BepInEx;
+using BepInEx.Bootstrap;
+using Cysharp.Threading.Tasks;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using System.Text.RegularExpressions;
+using UnityEngine.Networking;
+
+namespace StationeersLaunchPad
+{
+    // similar to jixxed's updater
+    public static class LaunchPadUpdater
+    {
+        public static List<string> Assemblies = new List<string>()
+        {
+            "RG.ImGui",
+            "StationeersMods.Interface",
+            "StationeersMods.Shared",
+            "StationeersLaunchPad",
+        };
+
+        private static Regex versionRegex = new Regex(@"""tag_name""\:\s""([V\d.]*)""", RegexOptions.Compiled | RegexOptions.IgnoreCase);
+        private static Regex downloadRegexClient = new Regex(@"""browser_download_url""\:\s""([^""]*StationeersLaunchPad-v.+\.zip)""", RegexOptions.Compiled | RegexOptions.IgnoreCase);
+        private static Regex downloadRegexServer = new Regex(@"""browser_download_url""\:\s""([^""]*StationeersLaunchPad-server-v.+\.zip)""", RegexOptions.Compiled | RegexOptions.IgnoreCase);
+
+        public static async UniTask CheckVersion()
+        {
+            var request = UnityWebRequest.Get("https://api.github.com/repos/StationeersLaunchPad/StationeersLaunchPad/releases/latest");
+            Logger.Global.Log($"Requesting version...");
+            var result = await request.SendWebRequest();
```

Comment:
is there a timeout set by default here? if not we probably should set a reasonable one so it doesn't just hang if they are on a questionable connection